clients/samples/ext_op/sample_hipblaslt_ext_op_matrixtransform.cpp

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (C) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "../common/helper.h"
+#include <algorithm>
+#include <hip/hip_runtime.h>
+#include <hipblaslt/hipblaslt.h>
+#include <iostream>
+#include <memory>
+#include <numeric>
+#include <vector>
+
+template <bool RowMaj>
+int64_t getLeadingDimSize(int64_t numRows, int64_t numCols)
+{
+    return RowMaj ? numCols : numRows;
+}
+
+void simpleTransformF32(int64_t           m,
+                        int64_t           n,
+                        bool              rowMajA,
+                        bool              rowMajB,
+                        bool              rowMajC,
+                        bool              transA,
+                        bool              transB,
+                        float*            a,
+                        float*            b,
+                        float*            c,
+                        hipblasLtHandle_t handle,
+                        hipStream_t       stream)
+{
+    int32_t                     batchSize   = 1;
+    int64_t                     batchStride = m * n;
+    float                       alpha       = 1.f;
+    float                       beta        = 0.f;
```

Comment:
seems all the configurtions for B doesn't have to apper in the following sample codes. Just need to set B as null?

Reply:
Good suggestion, the intention of the example is for transforming A only, will remove configurations for B in this example.